src/plugins/dev_tools/public/plugin.ts

```diff
@@ -125,10 +127,19 @@ export class DevToolsPlugin implements Plugin<DevToolsSetup> {
     };
   }
 
-  public start() {
+  public start(core: CoreStart) {
     if (this.getSortedDevTools().length === 0) {
       this.appStateUpdater.next(() => ({ navLinkStatus: AppNavLinkStatus.hidden }));
     }
+    core.chrome.navControls.registerRightNavigation({
+      // order of dev tool should be after advance settings
+      order: 2,
+      appId: this.id,
+      http: core.http,
+      application: core.application,
```

Comment:
A little bit weird why should plugins explicitly pass the props: `http` and `application` here. Should be accessible within RightNavigation component I suppose.

Reply:
Would it be easy for component to get core dependency?

Reply:
I am thinking what if RightNavigation requires more props from core, should all the registeration adjust accordingly?

Reply:
But this need to set osdReact as dependency? I think it may not be good to set core module depend on this.

Reply:
What about passing the coreStart to the component inside chrome_service instead of asking plugins to pass in such params?

Reply:
Sure. That makes sense.